substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1872,6 +1883,33 @@ pub mod pallet {
 			ensure_signed(origin)?;
 			Self::do_payout_stakers_by_page(validator_stash, era, page)
 		}
+
+		/// Migrates an account's `RewardDestination::Controller` to
+		/// `RewardDestination::Account(controller)`.
+		///
+		/// Effects will be felt instantly (as soon as this function is completed successfully).
+		///
+		/// This will waive the transaction fee if the `payee` is successfully migrated.
+		#[pallet::call_index(27)]
+		#[pallet::weight(T::WeightInfo::update_payee())]
+		pub fn update_payee(
+			origin: OriginFor<T>,
+			controller: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			let ledger = Self::ledger(StakingAccount::Controller(controller.clone()))?;
+
+			#[allow(deprecated)]
+			if Payee::<T>::get(&ledger.stash) != RewardDestination::Controller {
+				return Ok(Pays::Yes.into())
+			}
```

Comment:
nit: I think it would be better if we returned an error here just to signal to the user that their parameters were wrong and no state change took place, the fee gets paid too. Would also require a small tweak to the UT.

```suggestion
			ensure!(
				(Payee::<T>::get(&ledger.stash) == {
					#[allow(deprecated)]
					RewardDestination::Controller
				}),
				Error::<T>::NotController
			);
```

Reply:
Resolved in [89a1ed1](https://github.com/paritytech/polkadot-sdk/pull/2380/commits/89a1ed179d25e222eb0cd9d33b875d8f5c1d8af1)